pytorch_lightning/core/hooks.py

```diff
@@ -805,6 +815,18 @@ def on_after_batch_transfer(self, batch, dataloader_idx):
         """
         return batch
 
+    @property
+    def prepare_data_per_node(self) -> bool:
+        """
+        If True, each LOCAL_RANK=0 will call prepare data.
+        Otherwise only NODE_RANK=0, LOCAL_RANK=0 will prepare data.
+        """
+        return self._prepare_data_per_node
+
+    @prepare_data_per_node.setter
+    def prepare_data_per_node(self, prepare_data_per_node: bool) -> None:
+        self._prepare_data_per_node = prepare_data_per_node
```

Comment:
Why don't we simply have

```python
def __init__():
    self.prepare_data_per_node = True
```

If we don't need custom logic in the getter/setter nor to make it read-only?

Reply:
good point. we could simplify this @ninginthecloud